promql/bench_test.go

```diff
@@ -131,6 +135,9 @@ func BenchmarkRangeQuery(b *testing.B) {
 		{
 			expr: "label_join(a_X, 'l2', '-', 'l', 'l')",
 		},
+		{
+			expr: "max_over_time(a_X[10s])",
```

Comment:
This is duplicate with the rate() tests.

Reply:
It's not, actually. It was meant as a benchmark of the reuse logic in this PR (to test whether there was significant slowdown vs. the previous implementation when there was no actual overlap between steps and thus the newly added code was purely overhead. (There was no significant difference, even though it doesn't appear in the benchmark outputs I had attached above.)

Since that was specific to this PR though, I've now dropped it altogether.